lib/galaxy/webapps/galaxy/controllers/history.py

```diff
@@ -184,28 +184,31 @@ def make_private(self, trans, history_id=None, all_histories=False, **kwd):
             if history:
                 histories.append(history)
         if not histories:
-            return self.message_exception(trans, "Invalid history or histories specified.")
+            raise exceptions.RequestParameterMissingException("No history or histories specified.")
         private_role = trans.app.security_agent.get_private_user_role(trans.user)
         user_roles = trans.user.all_roles()
         private_permissions = {
             trans.app.security_agent.permitted_actions.DATASET_MANAGE_PERMISSIONS: [private_role],
             trans.app.security_agent.permitted_actions.DATASET_ACCESS: [private_role],
         }
         for history in histories:
-            self.history_manager.error_unless_mutable(history)
-            # Set default role for history to private
-            trans.app.security_agent.history_set_default_permissions(history, private_permissions)
-            # Set private role for all datasets
-            for hda in history.datasets:
-                if (
-                    not hda.dataset.library_associations
-                    and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
-                    and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
-                ):
-                    # If it's not private to me, and I can manage it, set fixed private permissions.
-                    trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
-                    if not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset):
-                        raise exceptions.InternalServerError("An error occurred and the dataset is NOT private.")
+            try:
+                # Set default role for history to private
+                trans.app.security_agent.history_set_default_permissions(history, private_permissions)
+                # Set private role for all datasets
+                for hda in history.datasets:
+                    if (
+                        not hda.dataset.library_associations
+                        and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
+                        and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
+                    ):
+                        # If it's not private to me, and I can manage it, set fixed private permissions.
+                        trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
+                        if not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset):
+                            raise exceptions.InternalServerError("An error occurred and the dataset is NOT private.")
```

Comment:
This seems pretty odd, is this code path reachable after calling `trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)` ?

Reply:
if it's unreachable we could remove the outer blanket exception handling, which contradicts the message we return on line 213

Reply:
I didn't check, this has been like that since the beginning around 6 years ago (https://github.com/galaxyproject/galaxy/pull/6952). Looking at the implementation of `set_all_dataset_permissions` now it seems like this check is indeed redundant. I can drop it as a small perf improvement.